packages/sd-jwt/src/action-handler.ts

```diff
@@ -175,22 +190,37 @@ export class SDJwtPlugin implements IAgentPlugin {
   async verify(sdjwt: SDJwtVcInstance, context: IRequiredContext, data: string, signature: string) {
     const decodedVC = await sdjwt.decode(`${data}.${signature}`)
     const issuer: string = ((decodedVC.jwt as Jwt).payload as Record<string, unknown>).iss as string
-    if (!issuer.startsWith('did:')) {
-      throw new Error('invalid_issuer: issuer must be a did')
+    const header = (decodedVC.jwt as Jwt).header as Record<string, any>
+    const x5c: string[] | undefined = header?.x5c as string[]
+    let jwk: JWK | JsonWebKey | undefined = undefined
+    let key: IKey | undefined
+    if (issuer.includes('did:')) {
+      const identifier: IIdentifier = await context.agent.didManagerGet({ did: issuer.split('#')[0] })
+      key = await context.agent.getKey({ identifier })
```

Comment:
Huh? why are we asking the agent again for an identifier it controls? We should be doing DID resolution for the DID and then we should be matching the kid from the SD-JWT header to get the correct VM and thus public key

Reply:
The code you have been replacing was literally doing that btw

Reply:
reverted the code that was doing the did resolution